### PR TITLE
Add rule

### PR DESCRIPTION
**Pull request by APIYOJENNIFER**
Issue #1 

- Add a rule
- Handle Field and operator select options
- Handle validation of input fields
- Handle deletion of a rule

### CODE REVIEW

File: src/components/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+interface ErrorProps {
+  isValid;
```
Comment (ThatcherK) [inline src/components/Error.tsx]: Why don't these props have types ideally that would render the interface useless? For example `isValid: boolean` and so forth. Or is this a new syntax I don't know about?

File: src/components/GeneralButton.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+interface ButtonProps {
+  className: string;
+  buttonText: string;
+  onClick: () => void;
+}
+
+const GeneralButton = ({ ...props }: ButtonProps) => (
```
Comment (ThatcherK) [inline src/components/GeneralButton.tsx]: Include return type for the component

File: src/components/Logical.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { logicalOperators } from '../utils';
+
+interface LogicalProps {
+  onLogicalChange: (logical: string) => void;
+}
+
+const Logical = ({ ...props }: LogicalProps) => {
```
Comment (ThatcherK) [inline src/components/Logical.tsx]: Include component return type

File: src/components/Rule.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import RuleItem from './RuleItem';
+
+interface RuleProps {
+  rules: {
```
Comment (ThatcherK) [inline src/components/Rule.tsx]: Here you could have an interface called Rule  and then use it inside RuleProps like this `interface RuleProps {rules:Rule[]}`.That makes it easier to re-use `Rule` anywhere else in the code base

**Review by ThatcherK — CHANGES_REQUESTED** [review]
Am getting this error when I run the application
Screenshot 2023-06-07 at 15 24 23

File: src/components/Error.tsx (hunk 1 above)
Comment (APIYOJENNIFER) [reply src/components/Error.tsx]: I had instances where the type would be inferred and I didn't need to explicitly annotate it but it's not the case here. Let me fix it

**Review by ThatcherK — APPROVED** [review]

**Pull request merged**